### PR TITLE
Added LZ4 compression

### PR DESCRIPTION
**Pull request by railsmechanic**

### CODE REVIEW

**Comment by mholt** [comment]
Cool, thanks for your contribution!

File: archiver_test.go [hunk 1]
```diff
@@ -74,7 +74,7 @@ func symmetricTest(t *testing.T, name string, ar Archiver) {
 
 		if info.IsDir() {
 			// stat dir instead of read file
-			_, err := os.Stat(origPath)
+			_, err = os.Stat(origPath)
```
Comment (shurcooL) [inline archiver_test.go]: Just a nitpicky question out of curiosity, what was the motivation/rationale behind this change?

Comment (railsmechanic) [inline archiver_test.go]: My IDE indicates that this `err` is shadowed by the definition at line 70. 😬

Comment (mholt) [inline archiver_test.go]: I noticed this, but it looks harmless in this case.

Comment (railsmechanic) [inline archiver_test.go]: Absolutely! A harmless change...

Comment (shurcooL) [inline archiver_test.go]: I see, thanks.

I would want to discuss whether a shadowed variable with name `err` and type `error` is something that warrants issuing warnings about, but that's a discussion to have elsewhere.

Comment (mholt) [inline archiver_test.go]: I think it depends largely on the context. That might be why go vet doesn't issue warnings about it...

Comment (railsmechanic) [inline archiver_test.go]: It's great being a software developer, where a single colon is a starting point for a conversation. 👍

**Pull request merged**